server/neptune/workflows/internal/deploy/workflow.go

```diff
@@ -55,7 +58,55 @@ func Workflow(ctx workflow.Context, request Request) error {
 	runner := newRunner(ctx, request)
 
 	// blocking call
-	return runner.Run(ctx)
+	err := runner.Run(ctx)
+	if err != nil {
+		return errors.Wrap(err, "executing deploy workflow")
+	}
+
+	// Spin up a child workflow to handle Terraform operations
+	childWorkflowOptions := workflow.ChildWorkflowOptions{
+		TaskQueue: terraform.TaskQueue,
+		//TODO: match workflow id to format used by deploy workflow
+		WorkflowID: fmt.Sprintf("%s-%s", request.Repository.FullName, request.Root.Name),
+	}
+	ctx = workflow.WithChildOptions(ctx, childWorkflowOptions)
+	terraformWorkflowRequest := TerraformRequest{
+		Repo: terraform.Repo{
+			FullName: request.Repository.FullName,
+			Owner:    request.Repository.Owner,
+			Name:     request.Repository.Name,
+			URL:      request.Repository.URL,
+		},
+		Root: terraform.Root{
+			Name: request.Root.Name,
+			Apply: terraform.Job{
+				Steps: convertToTerraformSteps(request.Root.Apply.Steps),
+			},
+			Plan: terraform.Job{
+				Steps: convertToTerraformSteps(request.Root.Plan.Steps),
+			},
+		},
+	}
+	err = workflow.ExecuteChildWorkflow(ctx, terraform.Workflow, terraformWorkflowRequest).Get(ctx, nil)
+	if err != nil {
+		return errors.Wrap(err, "executing child terraform workflow")
+	}
+	return nil
```

Comment:
Why is this here? this should be happening for each revision in the queue worker